qiskit/primitives/containers/data_bin.py

```diff
@@ -54,10 +63,33 @@ def __repr__(self):
         vals = (f"{name}={getattr(self, name)}" for name in self._FIELDS if hasattr(self, name))
         return f"{type(self)}({', '.join(vals)})"
 
+    def __getitem__(self, key: str) -> Any:
+        if not hasattr(self, key):
+            raise KeyError(f"Key ({key}) does not exist in this data bin.")
+        return getattr(self, key)
+
+    def __contains__(self, key: str) -> bool:
+        return hasattr(self, key)
+
+    def __iter__(self) -> Iterable[str]:
+        return iter(self._FIELDS)
+
+    def keys(self) -> list[str]:
+        """Return a list of field names."""
+        return self._FIELDS
```

Comment:
I'm not sure if FIELDS is guaranteed to be a tuple rather than al ist if a user creates and initializes their own databin, so if not might be good to force it
```suggestion
        return tuple(self._FIELDS)
```